### PR TITLE
allow cross repo workflow trigger (#526)

### PR DESCRIPTION
**Pull request by yuunlimm**
### description 

this is to allow aptos-core repo to trigger processor intregration test to support a case where json files are updated in aptos-core, we pass in commit hash along with the event, and it updates dependency using toml-cli. 

### Test Plan
PR level test pass: https://github.com/aptos-labs/aptos-indexer-processors/actions/runs/11131978344
dispatch passed: https://github.com/aptos-labs/aptos-indexer-processors/actions/runs/11131969466

### CODE REVIEW

File: .github/workflows/integration-tests.yaml [hunk 1]
```diff
@@ -1,25 +1,59 @@
 name: Run Integration Tests
 
 on:
+  repository_dispatch:
+    types: [json-change-detected]  # Custom event type to trigger the workflow
```
Comment (rtso) [inline .github/workflows/integration-tests.yaml:5]: nit: is it possible to make this more specific?

File: .github/workflows/integration-tests.yaml [hunk 2]
```diff
@@ -1,25 +1,59 @@
...
+
+      # Update aptos-system-utils dependency using toml-cli
+      - name: Update aptos-system-utils dependency
+        if: ${{ github.event_name == 'repository_dispatch' || github.event_name == 'workflow_dispatch' }}
+        run: |
+          echo "Updating aptos-system-utils dependency in Cargo.toml to use commit hash ${{ github.event.inputs.commit_hash }}"
+          toml set rust/Cargo.toml workspace.dependencies.aptos-system-utils.rev "${{ github.event.inputs.commit_hash }}" > Cargo.tmp && mv Cargo.tmp rust/Cargo.toml
+
+      # Update aptos-indexer-test-transactions dependency using toml-cli
+      - name: Update aptos-system-utils dependency
```
Comment (rtso) [inline .github/workflows/integration-tests.yaml:41]: forgot to update name

File: .github/workflows/integration-tests.yaml [hunk 3]
```diff
@@ -1,25 +1,59 @@
...
+        if: ${{ github.event_name == 'repository_dispatch' || github.event_name == 'workflow_dispatch' }}
+        run: |
+          echo "Updating aptos-system-utils dependency in Cargo.toml to use commit hash ${{ github.event.inputs.commit_hash }}"
```
Comment (rtso) [inline .github/workflows/integration-tests.yaml:44]: forgot to update this

**Review by rtso — APPROVED** [review]

File: .github/workflows/integration-tests.yaml (hunk 1 above)
Comment (yuunlimm) [reply .github/workflows/integration-tests.yaml:5]: how specific are you thinking? I think we can though, it's all custom

Comment (rtso) [reply .github/workflows/integration-tests.yaml:5]: maybe something like `test-txn-json-change-detected`?

**Pull request merged**